src/Utilities/for_testing/checkMatrix.hpp

```diff
@@ -50,9 +51,13 @@ struct CheckMatrixResult
  *                         left block of b_mat.
  *  \param[in] b_mat     - the matrix to check
  *  \param[in] check_all - if true continue to check matrix elements after failure
+ *  \param[in] eps       - add a tolerance for Catch Approx checks. Default to same as in Approx. 
  */
 template<class M1, class M2>
-CheckMatrixResult checkMatrix(M1& a_mat, M2& b_mat, const bool check_all = false)
+CheckMatrixResult checkMatrix(M1& a_mat,
+                              M2& b_mat,
+                              const bool check_all              = false,
+                              const double eps = std::numeric_limits<float>::epsilon() * 100)
```

Comment:
I don't feel right with this hidden choice. Could you use std::optional and default to `std::nullopt_t`

Reply:
Think I have this working with optional.